wideq/core.py

```diff
@@ -169,68 +196,146 @@ def __init__(self, device_id, code):
         self.code = code
 
 
-API_ERRORS = {
+API2_ERRORS = {
     "0102": NotLoggedInError,
     "0106": NotConnectedError,
-    "0100": FailedRequestError,
-    9000: InvalidRequestError,  # Surprisingly, an integer (not a string).
 }
 
 
-def lgedm_post(url, data=None, access_token=None, session_id=None):
-    """Make an HTTP request in the format used by the API servers.
-
-    In this format, the request POST data sent as JSON under a special
-    key; authentication sent in headers. Return the JSON data extracted
-    from the response.
+def thinq2_headers(
+    extra_headers={},
+    access_token=None,
+    user_number=None,
+    country=DEFAULT_COUNTRY,
+    language=DEFAULT_LANGUAGE,
+):
+    """Prepare API2 header."""
 
-    The `access_token` and `session_id` are required for most normal,
-    authenticated requests. They are not required, for example, to load
-    the gateway server data or to start a session.
-    """
     headers = {
-        'x-thinq-application-key': APP_KEY,
-        'x-thinq-security-key': SECURITY_KEY,
-        'Accept': 'application/json',
+        "Accept": "application/json",
+        "Content-type": "application/json;charset=UTF-8",
+        "x-api-key": V2_API_KEY,
+        "x-client-id": V2_CLIENT_ID,
+        "x-country-code": country,
+        "x-language-code": language,
+        "x-message-id": V2_MESSAGE_ID,
+        "x-service-code": SVC_CODE,
+        "x-service-phase": V2_SVC_PHASE,
+        "x-thinq-app-level": V2_APP_LEVEL,
+        "x-thinq-app-os": V2_APP_OS,
+        "x-thinq-app-type": V2_APP_TYPE,
+        "x-thinq-app-ver": V2_APP_VER,
+        "x-thinq-security-key": SECURITY_KEY,
     }
+
     if access_token:
-        headers['x-thinq-token'] = access_token
-    if session_id:
-        headers['x-thinq-jsessionId'] = session_id
-
-    with retry_session() as session:
-        res = session.post(url, json={DATA_ROOT: data}, headers=headers)
-    out = res.json()[DATA_ROOT]
-
-    # Check for API errors.
-    if 'returnCd' in out:
-        code = out['returnCd']
-        if code != '0000':
-            message = out['returnMsg']
-            if code in API_ERRORS:
-                raise API_ERRORS[code](code, message)
-            else:
-                raise APIError(code, message)
-
-    return out
-
-
-def oauth_url(auth_base, country, language):
-    """Construct the URL for users to log in (in a browser) to start an
-    authenticated session.
-    """
+        headers["x-emp-token"] = access_token
+
+    if user_number:
+        headers["x-user-no"] = user_number
```

Comment:
Is the user number really necessary to make the API work? A lot of things could be simplified if it could be omitted, and it seems strange that the API would need it in addition to the token.

Reply:
Yes it is, I'm sure about this